homeassistant/components/upnp/__init__.py

```diff
@@ -179,8 +191,8 @@ def _substitute_hass_ports(ports, hass_port=None):
     # set up port mapping
     if conf.get(CONF_ENABLE_PORT_MAPPING):
         _LOGGER.debug("Enabling port mapping")
-        local_ip = domain_data["local_ip"]
-        ports = conf.get("ports", {})
+        local_ip = domain_data[CONF_LOCAL_IP]
+        ports = conf.get(CONF_PORTS)
```

Comment:
How is the config entry data created when the config flow is triggered via discovery? Is there a default value set here for ports?